### PR TITLE
feat: use correct ARIA roles/attributes for drawer & modal

### PR DESCRIPTION
**Pull request by teevik**

### CODE REVIEW

**Comment by teevik** [comment]
Should we also use `aria-labelledby` and `aria-describedby`? (https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html#rps_label)

Where `aria-labelledby` would point to the `ModalTitle` element, and `aria-describedby` would point to the `ModalBody` element. 

---

I think we would need to make the context provider give out an id for both. And then add that id to it's respective element. 

The issue is the user could have multiple `ModalBody`, so then multiple elements would have the same id. Or there could be no `ModalBody`, and then the `aria-describedby` would point to nothing. I haven't found out if that's bad though. Screen readers probably just ignore it

**Comment by resir014** [comment]
> Should we also use aria-labelledby and aria-describedby? Where aria-labelledby would point to the ModalTitle element, and aria-describedby would point to the ModalBody element.

I see [reactstrap](https://reactstrap.github.io/) use it in their modal. Maybe on an optional prop titled `labelledBy`.

As for the multiple `ModalBody` issue, yeah, people could do that, although as a guideline there can only be one ModalBody, so let's assume that people have only one `ModalBody`.

File: packages/drawer/src/components/Drawer.tsx [hunk 1]
```diff
@@ -127,6 +134,10 @@ class Drawer extends React.Component<DrawerProps, DrawerState> {
                   this.state.isOpen ? 'is-open' : 'is-closed',
                   this.props.className
                 )}
+                role="dialog"
+                aria-modal="true"
+                aria-labelledby={this.ids.labelledBy}
+                aria-describedby={this.ids.describedBy}
```
Comment (resir014) [inline packages/drawer/src/components/Drawer.tsx]: We only need `aria-labelledby` for modal/dialog boxes. See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

File: packages/drawer/src/components/Drawer.tsx [hunk 2]
```diff
@@ -35,6 +35,12 @@ export interface DrawerState {
 }
 
 class Drawer extends React.Component<DrawerProps, DrawerState> {
+  private identifier = performance.now().toString(); // Generates unique values
+  private ids = {
+    labelledBy: this.identifier + '-label',
+    describedBy: this.identifier + '-body'
+  };
```
Comment (resir014) [inline packages/drawer/src/components/Drawer.tsx]: I'd rather we can set `labelledBy` through props, if that's possible.

File: packages/drawer/src/components/DrawerContext.tsx [hunk 3]
```diff
@@ -2,6 +2,10 @@ import * as React from 'react';
 
 interface DrawerContextAPI {
   overflow: boolean;
+  ids: {
+    labelledBy: string;
+    describedBy: string;
+  };
```
Comment (resir014) [inline packages/drawer/src/components/DrawerContext.tsx]: Same as above, set `labelledBy` through props.

**Comment by resir014** [comment]
> Should i also implement a way to add id to the ModalTitle?

I think those should just be passed to the context by the props, and have it be derive from it, no?

**Comment by teevik** [comment]
> > Should i also implement a way to add id to the ModalTitle?
> 
> I think those should just be passed to the context by the props, and have it be derive from it, no?

Not sure i understand, so we would use it like `<Modal labelledBy="someId">` and then the string provided would be used for two things?
* Used for aria tag: `<ModalWrapper aria-labelledby={labelledBy}>`
* Id would get passed down in context and used in ModalTitle as id: `<ModalTitle id={labelledBy}>`

**Comment by resir014** [comment]
> [...] so we would use it like <Modal labelledBy="someId"> and then the string provided would be used for two things?

Correct, is that possible?

**Comment by teevik** [comment]
Yep

**Comment by teevik** [comment]
^ That's the implementation
Not sure if that's what you meant though

**Comment by resir014** [comment]
Yes, exactly! 👍

**Pull request merged**